packages/palette_generator/lib/palette_generator.dart

```diff
@@ -1106,8 +1106,56 @@ class _ColorCutQuantizer {
 
     final ByteData imageData =
         await image.toByteData(format: ui.ImageByteFormat.rawRgba);
+    if(filters==null)
+      return await compute(_quantizefromByte, {
+        "byteData":imageData,
+        "maxColors":maxColors,
+        "width":image.width,
+        "height":image.height,
+        "region":region,
+        "paletteColors":_paletteColors,
+        "filters":null
+        });
+    else
+      return _quantizefromByte({
+       "byteData":imageData,
+        "maxColors":maxColors,
+        "width":image.width,
+        "height":image.height,
+        "region":region,
+        "paletteColors":_paletteColors,
+        "filters":filters 
+      });
+  }
+
+  static List<PaletteColor> _quantizefromByte(Map map){
+    ByteData bd = map["byteData"];
+    int maxColors = map["maxColors"];
+    int width = map["width"];
+    int height = map["height"];
+    Rect region = map["region"];
+    List<PaletteFilter> filters = map["filters"];
+    List<PaletteColor> _paletteColors =map["paletteColors"];
+    
+
+    const int quantizeWordWidth = 5;
+    const int quantizeChannelWidth = 8;
+    const int quantizeShift = quantizeChannelWidth - quantizeWordWidth;
+    const int quantizeWordMask =
+        ((1 << quantizeWordWidth) - 1) << quantizeShift;
+
+    Color quantizeColor(Color color) {
```

Comment:
Don't duplicate code: make `quantizeColor` a `static`, private function.

You can move the `quantizeWordMask` definition (and dependent values) inside of the function.